src/main.ts

```diff
@@ -28,6 +28,20 @@ const apiConfig = fetchRetry<Partial<Configuration>>(
     return new Configuration(undefined);
   });
 
-platformBrowserDynamic([{ provide: API_CONFIG, useValue: apiConfig }])
-  .bootstrapModule(AppModule)
-  .catch((err) => console.error(err));
+// Await page load
+const domContentLoadedPromise = new Promise((resolve) =>
+  document.addEventListener("DOMContentLoaded", () => {
+    document.removeEventListener("DOMContentLoader", () => {});
+    resolve();
+  })
+);
+
+// Bootstrap Angular
+Promise.all([apiConfigPromise, domContentLoadedPromise]).then((result) => {
+  const apiConfig: Configuration = result[0];
+
+  // this provider should mirror that in server.ts
+  platformBrowserDynamic([{ provide: API_CONFIG, useValue: apiConfig }])
+    .bootstrapModule(AppModule)
+    .catch((err) => console.error(err));
+});
```

Comment:
As talked about, may be possible to transfer app config when moving from SSR to browser. Create a tracking issue for the backlog.

Reply:
https://github.com/QutEcoacoustics/workbench-client/issues/396